### ISSUE TITLE
The program gets stuck at "Receiving from tcp://0.tcp.ap.ngrok.io:12560"

### ISSUE THREAD

**Issue by shadow-knights** [alievk/avatarify#240]
![image](https://user-images.githubusercontent.com/32615509/86490803-631cb000-bd71-11ea-9b48-f353273332df.png)
A little help please

**Comment by mintmaker** [comment]
Probably the server is not running. Make sure it's running and you can connect to it properly (note: Sending/Receiving to/from just prints out your arguments; Log lines beginning with capital letters (e.g. SEND) show the actual sending/receiving process)

**Comment by ralcant** [comment]
Same here :( Anyone found any solution? My server is running and I checked in ngrok and the tunnels are also active, so i don't know what else to try :/

**Comment by bobvo23** [comment]
Are you using your own pc at your home network?
Is there any potential firewall that prevents the communications between colab and your PC?

**Comment by justinjohn0306** [comment]
> ![image](https://user-images.githubusercontent.com/32615509/86490803-631cb000-bd71-11ea-9b48-f353273332df.png)
> A little help please

Go to ngork dashboard and click on reset token and copy the new token to colab

**Comment by ivanvick-vick** [comment]
Any resolve on this yet?

**Comment by vrmoonghost** [comment]
I had the same problem but I got it to work unplugged  network cable did  it over wifi and allways worked  hope this helps not sur why maybe being blocked by my router

**Comment by ochen1** [comment]
https://github.com/demotomohiro/remocolab/issues/79#issuecomment-735436649

I think Google Colab blocked ngrok lol

We must find another TCP tunnel :(

**Comment by ochen1** [comment]
I found out how to fix it. I will soon be creating a new issue that explains this problem in depth.

**Comment by JohanAR** [comment]
Good work @ochen1 ! Can we close this issue? Sounds like this issue was outside the scope of Avatarify

**Comment by ochen1** [comment]
I am actually not sure of that. Remote GPU support is supposed to be covered by avatarify.

Also I have a really bad problem of procrastinating so I actually have a draft of the issue that will probably solve this but I haven't gotten the time to open that issue yet.

If it's urgent for anyone you guys can check the slack channel where I posted the solution temporarily so I don't forget.

Hopefully avatarify doesn't drop support for colab just because ngrock stopped working.

Either way I'm not the owner of avatarify so I can't make these decisions. I'm not even a contributor yet.

**Comment by ochen1** [comment]
I am actually not sure of that. Remote GPU support is supposed to be covered by avatarify.

Also I have a really bad problem of procrastinating so I actually have a draft of the issue that will probably solve this but I haven't gotten the time to open that issue yet.

If it's urgent for anyone you guys can check the slack channel where I posted the solution temporarily so I don't forget.

Hopefully avatarify doesn't drop support for colab just because ngrock stopped working.

Either way I'm not the owner of avatarify so I can't make these decisions. I'm not even a contributor yet.

**Comment by JohanAR** [comment]
Just meant that if colab is blocking something that is necessary for us to use it, that sounds like something that is out of our hands. But if you are able to help us work around that, then it would be great ofc. Personally I have no Idea if it would be possible to use Avatarify with colab without ngrok, but maybe someone else does

It's not urgent, I'm only trying to clean up old issues.

**Comment by ochen1** [comment]
I see, thank you. I noticed a lot of issues being closed. I think keeping this one open is reasonable.

There _is_ a solution to this.

**Comment by jjbecomespheh** [comment]
> I am actually not sure of that. Remote GPU support is supposed to be covered by avatarify.
> 
> Also I have a really bad problem of procrastinating so I actually have a draft of the issue that will probably solve this but I haven't gotten the time to open that issue yet.
> 
> If it's urgent for anyone you guys can check the slack channel where I posted the solution temporarily so I don't forget.
> 
> Hopefully avatarify doesn't drop support for colab just because ngrock stopped working.
> 
> Either way I'm not the owner of avatarify so I can't make these decisions. I'm not even a contributor yet.

Hi @ochen1 , can I check how can I access the slack channel where you posted the solution temporarily?

**Comment by ochen1** [comment]
> Hi @ochen1 , can I check how can I access the slack channel where you posted the solution temporarily?

See #49.

**Comment by alievk** [comment]
@ochen1 could you also post your solution here, so that people don't have to join Slack for that?

**Comment by ochen1** [comment]
@alievk I plan on posting on another issue soon... Just been recently busy a lot with a lot of time conflicts ⚔️

**Comment by eyaler** [comment]
i solved it in my colab by moving to cloudflared argo, which is also faster.

**Comment by alievk** [comment]
> i solved it in my colab by moving to cloudflared argo, which is also faster.

Cool! Would you share the solution?

**Comment by eyaler** [comment]
> 
> 
> > i solved it in my colab by moving to cloudflared argo, which is also faster.
> 
> Cool! Would you share the solution?

sure!
demo: https://twitter.com/eyaler/status/1356455805017591810
colab: j.mp/cam2head
repo: github.com/eyaler/avatars4all

**Comment by alievk** [comment]
Cool stuff, thanks!!

**Comment by ochen1** [comment]
Solution posted as #369.

**Comment by JohanAR** [comment]
Please verify if solution is working

**Comment by shadow-knights** [comment]
Verified issue solved
(I recommend restarting the PC and following those steps if they don't work at first time)

**Comment by PiusBrassTek** [comment]
> > ![image](https://user-images.githubusercontent.com/32615509/86490803-631cb000-bd71-11ea-9b48-f353273332df.png)
> > A little help please
> 
> Go to ngork dashboard and click on reset token and copy the new token to colab

I have done this, but its still stuck there
![DESKTOP-NM66GNO - TeamViewer 9_13_2023 9_19_43 AM (1)](https://github.com/alievk/avatarify-python/assets/117092879/48179000-ec35-4097-a1a4-e8d9ad4e5b69)

**Comment by sirlouis8** [comment]
This is the error message I got from Google Colab, please can someone help me
![Screenshot_20240813-093320](https://github.com/user-attachments/assets/dcbd5de3-6c8d-4329-8b4e-a775e0f40dcd)
